packages/core/src/node/backend-application.ts

```diff
@@ -169,7 +169,11 @@ export class BackendApplication {
 
         server.listen(port, hostname, () => {
             const scheme = this.cliParams.ssl ? 'https' : 'http';
-            this.logger.info(`Theia app listening on ${scheme}://${hostname || 'localhost'}:${server.address().port}.`);
+            const address = server.address();
+            if (typeof address === 'string') {
+                throw new Error('Unexpected pipe address.');
```

Comment:
Shouldn't we add what was the expected and the actual address? What's your take on that?

Reply:
Can you please explain why do you call it *pipe address*?

Reply:
they've fixed the typings in latest version and introduced the `AddressInfo` return type instead of the structure. but actually the lib was already like that, if you call `server.listen("/pipe/name/or/unix/sock")` you'd get a string as the result of `server.address()`.

also, I'm again trying to pin the version of `@types/node` to 8.